glotaran/builtin/megacomplexes/decay/util.py

```diff
@@ -24,11 +24,10 @@ def calculate_matrix(
 ):
 
     compartments = megacomplex.get_compartments(dataset_model)
-    initial_concentration = megacomplex.get_initial_concentration(dataset_model)
     k_matrix = megacomplex.get_k_matrix()
 
     # the rates are the eigenvalues of the k matrix
-    rates = k_matrix.rates(compartments, initial_concentration)
+    rates, _ = k_matrix.eigen(compartments)
```

Comment:
in glotaran/builtin/megacomplexes/decay/k_matrix.py 
> rates are the negatives of the eigenvalues
that's not in line with the comment or the expression here.
@s-weigand can you double check?

Reply:
True this could use some renaming... naming things ... `k_matrix_eigenvalues` 🙈
But since this is an internal and not a user-facing API we could maybe postpone this to a different issue?

Reply:
nope, you must use the `rates` method. This ensures that in a sequential model what @ism200 calls the "B matrix". This has numerical advantages as it avoids calculating the eigenvalues.

Reply:
@joernweissenborn How about we add an `eigenvalues` method that does the same as the old `rates` method and rates just returns `self.eigenvalues` ?

Reply:
I discussed it with @joernweissenborn and dropped the commit in favor of improving the comment.